src/main/app/src/pages/UserGate/index.tsx

```diff
@@ -1,27 +1,38 @@
-import React, { useState, useEffect } from 'react';
-import { useIdle, useEvent, useAsync } from 'react-use';
+import { UNAUTHORIZED } from 'http-status-codes';
 import { get } from 'lodash';
+import React, { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { UNAUTHORIZED } from 'http-status-codes';
-
-import { IDLE_TIMEOUT, ERROR_INTERNET_DISCONNECTED } from '#/src/constants';
-import { getMe } from '#/src/utils/api/getMe';
-import { isDev } from '#/src/utils';
-import useApiAsync from '#/src/hooks/useApiAsync';
+import { useAsync, useEvent, useIdle } from 'react-use';
+import {
+  ERROR_INTERNET_DISCONNECTED,
+  ERROR_KAYTTOOIKEUS_SERVICE,
+  IDLE_TIMEOUT,
+} from '#/src/constants';
 import AuthorizedUserContext from '#/src/contexts/AuthorizedUserContext';
-import SessionErrorModal from './SessionErrorModal';
 import { useHttpClient, useUrls } from '#/src/contexts/contextHooks';
+import useApiAsync from '#/src/hooks/useApiAsync';
+import { isDev } from '#/src/utils';
+import { getMe } from '#/src/utils/api/getMe';
+import AuthorizationErrorModal from './AuthorizationErrorModal';
 
 const UserGate = ({ fallback = null, children = null }) => {
   const apiUrls = useUrls();
   const httpClient = useHttpClient();
   const [isFocused, setFocused] = useState(true);
-  const [sessionErrorCode, setSessionErrorCode] = useState(null);
+  const [errorCode, setErrorCode] = useState<string | number | null>(null);
 
-  const { data } = useApiAsync({ promiseFn: getMe });
+  const { data, error: getMeError } = useApiAsync({ promiseFn: getMe });
+  const isLoaded = !!data?.oid;
```

Comment:
Itse koen tämän vähän luettavammaksi, mutta makuasia.
```suggestion
  const isLoaded = Boolean(data?.oid);
```